universal-application-tool-0.0.1/app/services/program/ProgramServiceImpl.java

```diff
@@ -177,8 +178,9 @@ public ProgramDefinition addBlockToProgram(
     Program program =
         programDefinition.toBuilder()
             .addBlockDefinition(blockDefinition)
+            .setLifecycleStage(LifecycleStage.DRAFT)
```

Comment:
(question): does this need to happen in each method?

If the program is a draft, then it remains a draft and it doesn't have to be specified.

If a program is published, then you shouldn't be able to make edits to them, but this will turn them back into a draft. Is that the intended behavior?